core/src/main/java/com/gorisse/thomas/sceneform/arcore/LightEstimate.kt

```diff
@@ -0,0 +1,263 @@
+    // based camera model with lux or lumen values.
+    // In order to keep the "standard" Filament behavior we scale AR Core values.
+    cameraExposureFactor: Float
+) = lightEstimate?.takeIf { it.state == LightEstimate.State.VALID }?.let { lightEstimate ->
+    when (config.mode) {
+        Config.LightEstimationMode.ENVIRONMENTAL_HDR ->
+            lightEstimate.environmentalHdrMainLightInfluence(
+                baseLight,
+                cameraExposureFactor,
+                config.environmentalHdrMainLightDirection,
+                config.environmentalHdrMainLightIntensity
+            )
+        else -> null
+    }
+}
+
+fun LightEstimate.environmentalHdrMainLightInfluence(
+    baseLight: Light,
+    // ARCore's light estimation uses unit-less (relative) values while Filament uses a physically
+    // based camera model with lux or lumen values.
+    // In order to keep the "standard" Filament behavior we scale AR Core values.
+    cameraExposureFactor: Float,
+    withDirection: Boolean = true,
+    withIntensity: Boolean = true
+): (Light.() -> Unit) = {
+    if (withDirection) {
+        environmentalHdrMainLightDirection.let { (x, y, z) ->
+            // TODO (or not):
+            //  If light is detected as shining up from below, we flip the Y
+            //  component so that we always end up with a shadow on the ground to
+            //  fulfill UX requirements.
+            direction = Direction(-x, -y, -z)
+        }
+    }
+    if (withIntensity) {
+        // Returns the intensity of the main directional light based on the inferred
+        // Environmental HDR Lighting Estimation. All return values are larger or equal to zero.
+        // The color correction method uses the green channel as reference baseline and scales the
+        // red and blue channels accordingly. In this way the overall intensity will not be
+        // significantly changed
+        environmentalHdrMainLightIntensity.let { (redIntensity, greenIntensity, blueIntensity) ->
+            // TODO : Do we have to apply the gray scale and/or convert to linear?
+            val colorIntensities = Float3(redIntensity, greenIntensity, blueIntensity) *
+                    cameraExposureFactor
+
+            max(colorIntensities).takeIf { it > 0 }?.let { maxIntensity ->
+                // Scale max r or b or g value and fit in range [0, 1)
+                // Note that if we were not using the HDR cubemap from ARCore for specular
+                // lighting, we would be adding a specular contribution from the main light
+                // here.
+                val colorIntensitiesFactors = (colorIntensities / maxIntensity)
+                color = baseLight.color * colorIntensitiesFactors
+                // TODO : Should we apply the intensity in addition to the color correction?
+//                intensity = baseLight.intensity *
+//                        colorIntensitiesFactors.toFloatArray().average().toFloat()
```

Comment:
@romainguy Do you think that we should apply the intensity to the `IndirectLight` if we already apply it to the colors?